ibis/backends/tests/test_client.py

```diff
@@ -661,3 +663,21 @@ def test_create_from_in_memory_table(con, t):
     finally:
         con.drop_table(tmp_name)
         assert tmp_name not in con.list_tables()
+
+
+@pytest.mark.duckdb
+def test_default_backend():
+    pytest.importorskip("duckdb")
+
+    df = pd.DataFrame({'a': [1, 2, 3]})
+    t = ibis.memtable(df)
+    expr = t.a.sum()
+    assert expr.execute() == df.a.sum()
```

Comment:
What happens when the user tries to execute an expression based on an unbound table?

Reply:
The execution will fail.

Reply:
The failure mode isn't great:

```
DBAPIError: (duckdb.Error) Invalid Error: AttributeError: 'function' object has no attribute '_export_to_c'
[SQL: SELECT t0.a
FROM t AS t0
 LIMIT ?]
[parameters: (10000,)]
(Background on this error at: https://sqlalche.me/e/14/dbapi)
```

Reply:
This is no longer relevant due to refactoring the code to handle the case where an expression isn't executable.